app/controllers/admin/customers_controller.rb

```diff
@@ -66,7 +66,12 @@ def collection
       return Customer.where("1=0") unless json_request? && params[:enterprise_id].present?
 
       Customer.of(managed_enterprise_id).
-        includes(:bill_address, :ship_address, user: :credit_cards)
+        includes(:bill_address, :ship_address, user: :credit_cards).
+        joins(:orders).
+        merge(Spree::Order.complete.not_state(:canceled)).
+        group("customers.id").
+        select("customers.*").
+        select("SUM(total - payment_total) AS balance_value")
```

Comment:
I think that this is correct as a refactor and even though it's maybe not the right value in all cases, it should be the same as before. That is something we can easily test with production data though.